[Linux] Web Inspector: missing memory measurements in memory timeline

carlosgcampos opened the pull request:
#### ec374dbfec091ed210b7534dd447e6e9e5262d1a
<pre>
[Linux] Web Inspector: missing memory measurements in memory timeline
<a href="https://bugs.webkit.org/show_bug.cgi?id=242500">https://bugs.webkit.org/show_bug.cgi?id=242500</a>

Reviewed by Adrian Perez de Castro and Patrick Angle.

In linux we don&apos;t show memory information for image, layers and page
categories. I don&apos;t think we can get image and layers, but at least we
can add others category with the memory not in javascript categories. We
can also not show the categories for which the memory is 0, because it&apos;s
confusing.

* Source/WebCore/page/linux/ResourceUsageThreadLinux.cpp:
(WebCore::ResourceUsageThread::platformCollectMemoryData): Set memory
for category other.
* Source/WebInspectorUI/UserInterface/Models/MemoryTimelineRecord.js:
(WI.MemoryTimelineRecord.memoryCategoriesFromProtocol): Do not show
categories for which the memory reported is 0.

Canonical link: <a href="https://commits.webkit.org/252496@main">https://commits.webkit.org/252496@main</a>
</pre>

patrickangle reviewed (approved):
Seems reasonable, although I do wonder if we should hide categories based on their actual support on each platform so we can continue to show "real" zeros on platforms that support those memory categories.

carlosgcampos committed:
[Linux] Web Inspector: missing memory measurements in memory timeline
https://bugs.webkit.org/show_bug.cgi?id=242500

Reviewed by Adrian Perez de Castro and Patrick Angle.

In linux we don't show memory information for image, layers and page
categories. I don't think we can get image and layers, but at least we
can add others category with the memory not in javascript categories. We
can also not show the categories for which the memory is 0, because it's
confusing.

* Source/WebCore/page/linux/ResourceUsageThreadLinux.cpp:
(WebCore::ResourceUsageThread::platformCollectMemoryData): Set memory
for category other.
* Source/WebInspectorUI/UserInterface/Models/MemoryTimelineRecord.js:
(WI.MemoryTimelineRecord.memoryCategoriesFromProtocol): Do not show
categories for which the memory reported is 0.

Canonical link: https://commits.webkit.org/252496@main
